contracts/AnchoredPriceView.sol

```diff
@@ -0,0 +1,331 @@
+                if (anchorRatioMantissa <= upperBoundAnchorRatio && anchorRatioMantissa >= lowerBoundAnchorRatio) {
+                    // only update and emit event if the source price is new, otherwise do nothing
+                    if (_prices[symbol] != sourcePrice) {
+                        _prices[symbol] = sourcePrice;
+                        emit PriceUpdated(symbol, sourcePrice);
+                    }
+                } else {
+                    emit PriceGuarded(symbol, sourcePrice, anchorPrice);
+                }
+            }
+        }
+    }
+
+    /**
+     * @notice Returns price denominated in USD, with 6 decimals
+     * @dev If price was posted by source, return it. Otherwise, return anchor price converted through source ETH price.
+     */
+    function prices(string calldata symbol) external view returns (uint256) {
+        uint256 price = _prices[symbol];
+
+        if (price != 0) {
+            return price;
+        } else {
+            uint256 usdPerEth = _prices["ETH"];
+            uint256 ethPerToken = anchor.getUnderlyingPrice(getCTokenAddress(symbol));
+
+            // ethPerToken has 18 decimals since the usdt, usdc, wbtc tokens hit
+            // usdPerEth has 6 decimals
+            // scaling by 1e18 and dividing leaves 1e6, as desired
+            return mul(usdPerEth, 1e18) / ethPerToken;
+        }
+    }
+
+    /**
+     * @dev fetch price in eth from proxy and convert to usd price using anchor usdc price.
+     * @dev Anchor usdc price has 30 decimals, and anchor general price has 18 decimals, so multiplying 1e18 by 1e18 and dividing by 1e30 yields 1e6
+     */
+    function getAnchorPrice(address tokenAddress, uint256 usdcPrice) public view returns (uint256) {
+        if ( tokenAddress == cUsdcAddress || tokenAddress == cUsdtAddress )  {
+            // hard code to 1 dollar
+            return oneDollar;
+        }
+
+        uint priceInEth = anchor.getUnderlyingPrice(tokenAddress);
+        uint additionalScale;
+        if ( tokenAddress == cWbtcAddress ){
+            // wbtc proxy price is scaled 1e(36 - 8) = 1e28, so we need 8 more to get to 36
+            additionalScale = 1e8;
+        } else {
+            // all other tokens are scaled 1e18, so we need 18 more to get to 36
+            additionalScale = 1e18;
+        }
+
+        // usdcPrice has 30 decimals, so final result has 6
+        return mul(priceInEth, additionalScale) / usdcPrice;
+    }
+
+    /**
+     * @notice Implements the method of the PriceOracle interface of Compound v2 and returns returns the Eth price for an asset.
+     * @dev converts from 1e6 decimals of Open Oracle to 1e(36 - underlyingDecimals) of PriceOracleProxy
+     * @param cToken The cToken address for price retrieval
+     * @return The price for the given cToken address
+     */
+    function getUnderlyingPrice(address cToken) external view returns (uint256) {
```

Comment:
I'm debating a bit if it's a good idea to return these in Ether versus simply returning them in USD, as we originally planned. This could probably drop some complexity (that we're going to add with other things like a staleness check)

Reply:
yeah, I say we keep ruminating on which numerator to return. I'm pretty comfortable with either, have code to do either, and glad to stay nimble about which we pick in the end